src/remat_data/dataset.py

```diff
@@ -122,5 +125,74 @@ def download_dataset(dataset_id: str):
         )
 
 
+
+# Each space is a typer option
+@spaces_app.command("upload", no_args_is_help=True)
+def upload_file(
+        Cure: bool = typer.Option(False, "--Cure", help=" Space: DSC Cure Kinetics"),
+        PostCure: bool = typer.Option(False, "--PostCure", help=" Space: DSC Post Cures"),
+        FrontVelocity: bool = typer.Option(False, "--FrontVelocity", help=" Space: Front velocities"),
+
+
+        dataset_name: Optional[str] = typer.Option(None, "--name", help="Optional name for the dataset"),
+        file_names: list[str] = typer.Argument(...)) -> None:
+    """
+       Upload given files to a specified space
+
+    This command uploads the files given in the arguments to the specified space_name.
+    A default dataset is created and the files are uploaded under the newly created dataset
+
+    Args:
+        Each space is a flag that specifies the space to upload the files to.
+        dataset_name (str): Optional name for the dataset. If not provided, a default name is used.
+        file_names List[str]: List of filenames to be uploaded. Files must be present at the directory where remat-data is run
+    Usage:
+       remat-data spaces upload --<space_name> --name <dataset_name> <file1> <file2>
+
+    Example:
+    remat-data spaces upload --DSC_Cure_Kinetics --name Test_Dataset test3.csv DSC_Curve.csv
+    Note:
+    - The space_name flag is mandatory. Use --help to see valid space names
+    - Files to be uploaded must be in the directory where remat-data is installed/run
+    """
+
+    # Keep this dictionary consistent with the space_map in config.py
+    space_names = {
+        "DSC Cure Kinetics": Cure,
+        "DSC Post Cures": PostCure,
+        "Front velocities": FrontVelocity,
+    }
+    space_name = next((name for name, value in space_names.items() if value), None)
```

Comment:
What happens if I specify `--Cure`  and `--PostCure`? It needs to make sure I don't specify multiple spaces.

Also, this is an interesting way to do the mapping, but it deserves a comment describing what is going on